constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -62,12 +62,12 @@ pub struct QuadraticSymbolicExpression<T: FieldElement, V> {
     /// Quadratic terms of the form `a * X * Y`, where `a` is a (symbolically)
     /// known value and `X` and `Y` are quadratic symbolic expressions that
     /// have at least one unknown.
-    quadratic: Vec<(Self, Self)>,
+    pub quadratic: Vec<(Self, Self)>,
```

Comment:
Can we keep the internal structure hidden? The main issue I have with this is that it allows you to mess with the fields directly. Not saying we will, but we cannot be sure that nobody does.

Reply:
I'm not at all against private fields, I just think not accessing them should be a linter rule instead of a compiler error 😉